HiP-DataStore.Model/Entity/ScoreBoard.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+
+namespace PaderbornUniversity.SILab.Hip.DataStore.Model.Entity
+{
+    public class ScoreBoard : SortedSet<ScoreRecord>
+    {
+       public ScoreBoard() : base(new UsersScoreComparer()) { }
+    }
+
+    //Making sorting first by scores. Then by Timestamp
+    class UsersScoreComparer : IComparer<ScoreRecord>
+    {
+        public int Compare(ScoreRecord pair1, ScoreRecord pair2)
+        {
+            int compareResult = pair1.Score.CompareTo(pair2.Score);
```

Comment:
As ReSharper suggests, the cases `pair1 == null` and `pair2 == null` should be handled